modules/light-clients/08-wasm/types/upgrade_test.go

```diff
@@ -98,536 +106,137 @@ func (suite *TypesTestSuite) TestVerifyUpgradeGrandpa() {
+			upgradedConsState = &types.ConsensusState{[]byte("consensus state")}
 
-// 			var tmUpgradedConsState exported.ConsensusState //nolint:gosimple // ignore error for test
-// 			tmUpgradedConsState = &ibctm.ConsensusState{
-// 				Timestamp:          time.Now(),
-// 				Root:               commitmenttypes.NewMerkleRoot([]byte("root hash")),
-// 				NextValidatorsHash: []byte("01234567890123456789012345678901"),
-// 			}
-// 			tmUpgradedConsStateBz, err := clienttypes.MarshalConsensusState(suite.chainA.App.AppCodec(), tmUpgradedConsState)
-// 			suite.Require().NoError(err)
+			tc.malleate()
 
-// 			upgradedConsState = &types.ConsensusState{
-// 				Data: tmUpgradedConsStateBz,
-// 			}
-// 			upgradedConsStateBz, err = clienttypes.MarshalConsensusState(suite.chainA.App.AppCodec(), upgradedConsState)
-// 			suite.Require().NoError(err)
+			proofUpgradedClient = []byte("proof upgraded client")
+			proofUpgradedConsState = []byte("proof upgraded consensus state")
 
-// 			tc.setup()
+			err = clientState.VerifyUpgradeAndUpdateState(
+				suite.ctx,
+				suite.chainA.Codec,
+				suite.store,
+				upgradedClient,
+				upgradedConsState,
+				proofUpgradedClient,
+				proofUpgradedConsState,
+			)
 
-// 			cs := suite.chainA.GetClientState(path.EndpointA.ClientID)
-// 			clientStore := suite.chainA.App.GetIBCKeeper().ClientKeeper.ClientStore(suite.chainA.GetContext(), path.EndpointA.ClientID)
+			expPass := tc.expErr == nil
+			if expPass {
+				suite.Require().NoError(err)
 
-// 			err = cs.VerifyUpgradeAndUpdateState(
-// 				suite.chainA.GetContext(),
-// 				suite.chainA.Codec,
-// 				clientStore,
-// 				upgradedClient,
-// 				upgradedConsState,
-// 				proofUpgradedClient,
-// 				proofUpgradedConsState,
-// 			)
+				// verify new client state and consensus state
+				clientStateBz := suite.store.Get(host.ClientStateKey())
+				suite.Require().NotEmpty(clientStateBz)
+				suite.Require().Equal(upgradedClient, clienttypes.MustUnmarshalClientState(suite.chainA.Codec, clientStateBz))
 
-// 			if tc.expPass {
-// 				suite.Require().NoError(err, "verify upgrade failed on valid case: %s", tc.name)
+				consStateBz := suite.store.Get(host.ConsensusStateKey(upgradedClient.GetLatestHeight()))
+				suite.Require().NotEmpty(consStateBz)
+				suite.Require().Equal(upgradedConsState, clienttypes.MustUnmarshalConsensusState(suite.chainA.Codec, consStateBz))
+			} else {
+				suite.Require().Error(err)
+				suite.Require().ErrorIs(err, tc.expErr)
 
-// 				clientState := suite.chainA.GetClientState(path.EndpointA.ClientID)
-// 				suite.Require().NotNil(clientState, "verify upgrade failed on valid case: %s", tc.name)
+				// verify client state is unchanged
+				clientState := clienttypes.MustUnmarshalClientState(suite.chainA.Codec, suite.store.Get(host.ClientStateKey()))
+				suite.Require().Equal(clientState, endpoint.GetClientState().(*types.ClientState))
```

Comment:
I think this is circular logic, `endpoint.GetClientState()` gets the client from the store. We would need to reference a client state var grabbed from state before the verify upgrade call

Reply:
regardless, I think we can remove this bit of code. An error returned to baseapp reverts all state changes anyways. We only need to check that state isn't changed on failure when we are using cached contexts and returning successful transaction to baseapp despite some failure route in the execution trace